api/api.go

```diff
@@ -809,6 +809,58 @@ func (api *Server) GetEvmTransfersByBlockHeight(ctx context.Context, in *iotexap
 	return &iotexapi.GetEvmTransfersByBlockHeightResponse{BlockEvmTransfers: transfers}, nil
 }
 
+// GetSystemLogByActionHash returns system log by action hash
+func (api *Server) GetSystemLogByActionHash(
+	ctx context.Context,
+	in *iotexapi.GetSystemLogByActionHashRequest) (*iotexapi.GetSystemLogByActionHashResponse, error) {
+	if !api.hasActionIndex || api.indexer == nil {
+		return nil, status.Error(codes.NotFound, blockindex.ErrActionIndexNA.Error())
+	}
+
+	h, err := hex.DecodeString(in.ActionHash)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	actIndex, err := api.indexer.GetActionIndex(h)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	sysLog, err := api.dao.GetSystemLog(actIndex.BlockHeight())
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	for _, log := range sysLog.ActionSystemLog {
+		if in.ActionHash == hex.EncodeToString(log.ActionHash) {
+			return &iotexapi.GetSystemLogByActionHashResponse{
+				ActionSystemLog: log,
+			}, nil
+		}
+	}
+	return nil, status.Errorf(codes.NotFound, "system log not found for action %s", in.ActionHash)
```

Comment:
I am not sure if this is the desired way to implement this on core end.

Reply:
yeah, i was originally think adding system log as part an action index `/blockindex/actionindex.go`
then GetByActionHash() becomes a direct DB access